cloud-aws-cloudformation/src/main/java/com/sequenceiq/cloudbreak/cloud/aws/AwsNetworkConnector.java

```diff
@@ -260,16 +260,18 @@ public void deleteNetworkWithSubnets(NetworkDeletionRequest networkDeletionReque
             AwsCredentialView credentialView = new AwsCredentialView(networkDeletionRequest.getCloudCredential());
             String regionName = networkDeletionRequest.getRegion();
             AmazonCloudFormationClient cfClient = awsClient.createCloudFormationClient(credentialView, regionName);
-            DeleteStackRequest deleteStackRequest = new DeleteStackRequest();
             String stackName = networkDeletionRequest.getStackName();
-            deleteStackRequest.setStackName(stackName);
-            cfClient.deleteStack(deleteStackRequest);
-            Waiter<DescribeStacksRequest> deletionWaiter = cfClient.waiters().stackDeleteComplete();
-            LOGGER.debug("CloudFormation stack deletion request sent with stack name: '{}' ", stackName);
-            DescribeStacksRequest describeStacksRequest = new DescribeStacksRequest().withStackName(stackName);
-            run(deletionWaiter, describeStacksRequest, null,
-                    String.format("Network delete failed (cloudformation: %s)", stackName),
-                    () -> awsCloudFormationErrorMessageProvider.getErrorReason(credentialView, regionName, stackName, ResourceStatus.DELETE_FAILED));
+            if (cfStackUtil.isCfStackExists(cfClient, stackName)) {
```

Comment:
If _isCfStackExists_ method can throw Retry.ActionFailedException, I think it should be called with retry (like in the AwsTerminateService)
`retryService.testWith2SecDelayMax15Times(() -> cfStackUtil.isCfStackExists(amazonCloudFormationClient, cFStackName));`

Reply:
you are right, thanks